### DISCUSSION TITLE
Polars dataframe / arrow table

### DISCUSSION THREAD

**Discussion by aduarterengifo** [perspective-dev/perspective#2169] [Ideas]
## Feature Request

Hi, I was wondering how much work it would be to add support for `polars.DataFrame`, I saw a pull request for `to_arrow()`  that seemed to have stalled.

**Comment by timkpaine** [comment]
In python and JS we already support instantiation from arrow (see python tests [here](https://github.com/finos/perspective/blob/master/python/perspective/perspective/tests/table/test_table_arrow.py)). We also support pulling in data in python from [numpy](https://github.com/finos/perspective/blob/master/python/perspective/perspective/core/data/np.py#L54) or [pandas](https://github.com/finos/perspective/blob/master/python/perspective/perspective/core/data/pd.py#L52). For export in python, you can go [to arrow](https://github.com/finos/perspective/blob/master/python/perspective/perspective/table/view.py#L421) or numpy / [pandas via numpy](https://github.com/finos/perspective/blob/master/python/perspective/perspective/table/view.py#L565). 

I think it would be fine to support importing/exporting to/from flat polars dataframes. We wouldn't want to introduce an extra required dependency, and it may be difficult to deconstruct a pivoted polars dataframe (as it is for pandas). It also might be straightforward to just import/export polars from the existing numpy/arrow APIs without having any first party support.

**Comment by aduarterengifo** [comment]
sorry for the duplicate.

this is the same as  #1157. 

I want to do this 
```
arrow_table = polars_df.to_arrow()
perspective_table = Table(arrow_table)
``` 

instead of 
```
arrow_table = polars_df.to_arrow()
stream = pa.BufferOutputStream()
writer = pa.RecordBatchStreamWriter(stream, arrow_table.schema)
writer.write_table(arrow_table)
writer.close()
perspective_table = Table(stream.getvalue().to_pybytes())
```
